### DISCUSSION TITLE
SwiftUI lifecycle

### DISCUSSION THREAD

**Discussion by mickeyl** [getsentry/sentry-cocoa#4001] [Q&A]
In a SwiftUI app, Sentry's automatic `UIKit` lifecycle breadcrumbs are pretty much useless.

How are you guys coping with that? Do you add custom breadcrumbs from all your top level `body` views or do you have a better ­– more automatic ­– approach?

**Comment by brustolin** [comment]
We dont have an automatic approach for SwiftUI, but you can use `SentryTracerView` from our SentrySwiftUI lib

**Reply by mickeyl** [reply]
If I read that correctly, `SentryTracedView` is more about measuring performance than emitting breadcrumbs. Can I get the latter one without having to pay for the performance analysis?

**Reply by brustolin** [reply]
No, not right now. 
You will have to come up with your own solution for registering breadcrumb for now.
